### ISSUE TITLE
Batch_size when using model.fit_generator()

### ISSUE THREAD

**Issue by HikaruL** [keras-team/keras#7661]
I'm trying to define a generator using mnist dataset: 
Here is my defined generator
![_20170815163337](https://user-images.githubusercontent.com/19523304/29341162-8b283c78-81d7-11e7-8706-52675853cade.png)

But when I run the script, it seems that the batch_size of samples fed into the model is not 128
    1/60000 [..............................] - ETA: 18707s - loss: 2.3394 - acc: 0.1172
    4/60000 [..............................] - ETA: 5729s - loss: 1.8231 - acc: 0.4160 
    7/60000 [..............................] - ETA: 3840s - loss: 1.5300 - acc: 0.5045
   10/60000 [..............................] - ETA: 3082s - loss: 1.3488 - acc: 0.5602
   13/60000 [..............................] - ETA: 2701s - loss: 1.2442 - acc: 0.6010
   16/60000 [..............................] - ETA: 2447s - loss: 1.1160 - acc: 0.6436

Not sure what went wrong. Thanks!

**Comment by GH1995** [comment]
```python
for a in range(128):
    batch_features[a] = x_train[a]
    batch_lables[a] = keras.utils.to_categorical(y_train[a], num_classes)
````
I guess the data you used is the same.

**Comment by HikaruL** [comment]
@GH1995 I modified the codes based on https://medium.com/@fromtheast/implement-fit-generator-in-keras-61aa2786ce98

**Comment by HikaruL** [comment]
@GH1995 I see what you mean by using the same data. But the output process line is still the same if I choose the random index from training data. Do you know how I can fix it?
